online-shop/src/Products/ProductDetails.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import { IProduct } from '../App';
+import '../Styles/ComponentsStyles/ProductDetails.scss';
+import ConfirmationModal from '../HelperComponents/ConfirmationModal';
+import { Redirect } from 'react-router-dom';
+
+interface IProps {
+  match?: any;
+  addProductToShoppingCartFunction : any;
+}
+
+interface IState {
+  toBeReceivedData: IProduct;
+  isLoading: boolean;
+  error?: Error;
+  isDeleteModalOpen: boolean;
+  shouldRedirectToShoppingCart: boolean;
+}
+
+export default class ProductDetails extends React.Component<IProps, IState> {
+  receivedProductData: IProduct = {
```

Comment:
Never use properties in React Components! Either use state or props. 

If you want to use props, then you need a callback function from the parent component to pass the data to it (and then get it back via props). Otherwise, just use the state.

Reply:
Basically in this case, you already have the data in the state, this property can easily be removed and the value from the state used instead.